### ISSUE TITLE
Could I get a code for preprocessing Voxceleb2?

### ISSUE THREAD

**Issue by kangyeolk** [shrubb/latent-pose-reenactment#6]
Hi, again!

I'm trying to train your model from scratch. And I preprocessed Voxceleb2 datasets as follows:

Following the preprocessing of `Neural Head Reenactment with Latent Pose Descriptors`
* [Sampling] Sample a frame at every [SAMPLE_RATE] = 25 (same as paper).
* [Resizing] Detect face image  -> make square -> enlarge 80% of a box -> resize 256 x 256
* [Detecting others] Extract facial landmarks / segmentation mask

All preprocessing codes were implemented referring your codes in ```utils/crop_as_in_dataset.py``` and ```utils/preprocess_dataset.sh```. Here, I have two questions about the preprocessing:

1 . I found that resizing operation generates rather unrealistic outputs due to the reflection padding. I think it could have a bad effect on training (in terms of perceptual loss). Also, segmentation masks based on these outputs are erroneous too. I attach intermediate results produced by your code :)

![individualImage](https://user-images.githubusercontent.com/26561999/107112717-fe907900-689c-11eb-8795-13ee6e08967f.png)

2. Second one is about sampling. I sampled every 25th frames (i.e., 0, 25, 50, ...) from the videos located at the same hash id. As you know, in that way, some hash ids have less than 9 frames. Did you discard those hash ids then..? I wonder the exact procedure to get lists: ```/data/splits/train.csv```, ```/data/splits/val.csv```

If possible, it would be greatly helpful if you can give a code for preprocessing. 
Thank you, Burkov!

**Comment by shrubb** [comment]
Hi,

[`LatentPoseFaceCropper` from utils/crop_as_in_dataset.py](https://github.com/shrubb/latent-pose-reenactment/blob/59629a64105c7c33fa01c461a3c65d3690f8533c/utils/crop_as_in_dataset.py#L202) should give you the full idea on how to preprocess. Everything you need is there. What is left is probably to parallelize (speed up) that code and to run it on VoxCeleb2.

Your preprocessing seems OK, with two minor comments:

- there's probably a mistake in the "80% box enlargement" step (your crops seem too tight) -- this is why you get unrealistic reflection padding;
- you don't have to use landmarks at all (this is one of the main points of our paper).

To your questions:

1. See my comment above. However, even with excessive reflection padding, training should still work. Of course, the system will learn to generate reflected foreheads which is wrong, but faces should still be OK. Your results actually seem fine, probably just need more iterations.
2. Good question. I just realized that we indeed had sequences with less than 9 frames. I solved it by resampling [here](https://github.com/shrubb/latent-pose-reenactment/blob/59629a64105c7c33fa01c461a3c65d3690f8533c/dataloaders/common/voxceleb.py#L170), allowing the "*Pose target*" to be present in "*Identity sources*" too. This probably isn't good. Thank you for enlightening me.

**Comment by enjoybo** [comment]
> Hi, again!
> 
> I'm trying to train your model from scratch. And I preprocessed Voxceleb2 datasets as follows:
> 
> Following the preprocessing of `Neural Head Reenactment with Latent Pose Descriptors`
> 
> * [Sampling] Sample a frame at every [SAMPLE_RATE] = 25 (same as paper).
> * [Resizing] Detect face image  -> make square -> enlarge 80% of a box -> resize 256 x 256
> * [Detecting others] Extract facial landmarks / segmentation mask
> 
> All preprocessing codes were implemented referring your codes in `utils/crop_as_in_dataset.py` and `utils/preprocess_dataset.sh`. Here, I have two questions about the preprocessing:
> 
> 1 . I found that resizing operation generates rather unrealistic outputs due to the reflection padding. I think it could have a bad effect on training (in terms of perceptual loss). Also, segmentation masks based on these outputs are erroneous too. I attach intermediate results produced by your code :)
> 
> ![individualImage](https://user-images.githubusercontent.com/26561999/107112717-fe907900-689c-11eb-8795-13ee6e08967f.png)
> 
> 1. Second one is about sampling. I sampled every 25th frames (i.e., 0, 25, 50, ...) from the videos located at the same hash id. As you know, in that way, some hash ids have less than 9 frames. Did you discard those hash ids then..? I wonder the exact procedure to get lists: `/data/splits/train.csv`, `/data/splits/val.csv`
> 
> If possible, it would be greatly helpful if you can give a code for preprocessing.
> Thank you, Burkov!

Hello, Can you reproduce the results of the paper now？I'm trying to train the model from scratch, but the training results are not so good.
